### PR TITLE
Allow setting AnyPointer fields with a capability

### PR DESCRIPTION
**Pull request by jparyani**
This is blocked on https://github.com/sandstorm-io/capnproto/pull/214

### CODE REVIEW

File: src/node-capnp/capnp.cc [hunk 1]
```diff
@@ -1280,23 +1280,39 @@ struct FromJsConverter {
...
-                                 buffer->size() / sizeof(capnp::word));
+        if (type.whichAnyPointerKind() == capnp::schema::Type::AnyPointer::Unconstrained::CAPABILITY) {
+          if (js->IsNull()) {
+            auto cap = capnp::Capability::Client(nullptr);
+            capnp::DynamicCapability::Client dynamicCap(kj::mv(cap));
+            return orphanage.newOrphanCopy(dynamicCap);
+          } else KJ_IF_MAYBE(cap, Wrapper::tryUnwrap<capnp::DynamicCapability::Client>(js)) {
+            return orphanage.newOrphanCopy(*cap);
+          } else if (!localCapType.IsEmpty()) {
+            // TODO(someday): handle schemaless local caps
```
Comment (kentonv) [inline src/node-capnp/capnp.cc]: Would it make sense to use InterfaceSchema(nullptr) here? (It's an interface with no methods.)

The idea is that the close() method would still work.

Maybe not very important?

Comment (jparyani) [inline src/node-capnp/capnp.cc]: I'm not sure how to make an InterfaceSchema from a nullptr, when i try to do `capnp::InterfaceSchema schema(nullptr)`, I get:
```
  /ekam-provider/canonical/node-capnp/capnp.cc:1294:38: error: no matching constructor for initialization of 'capnp::InterfaceSchema'
                capnp::InterfaceSchema schema(nullptr);
                                       ^      ~~~~~~~
  /ekam-provider/c++header/capnp/schema.h:438:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'nullptr_t' to 'const capnp::
    InterfaceSchema' for 1st argument
  class InterfaceSchema: public Schema {
        ^
  /ekam-provider/c++header/capnp/schema.h:438:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'nullptr_t' to 'capnp::
    InterfaceSchema' for 1st argument
  class InterfaceSchema: public Schema {
        ^
  /ekam-provider/c++header/capnp/schema.h:465:3: note: candidate constructor not viable: no known conversion from 'nullptr_t' to 'capnp::Schema' for 1st argument
    InterfaceSchema(Schema base): Schema(base) {}
```

Comment (kentonv) [inline src/node-capnp/capnp.cc]: Oh, actually, this should just work: `capnp::Schema::from<Capability>()`.

**Comment by jparyani** [comment]
Done

**Pull request merged**